Release 1.2.0-rc.2 (#4317)
opencontainers/runc: CHANGELOG.md

## [Unreleased]

## [1.2.0-rc.2] - 2024-06-26

> TRUE or FALSE, it's a problem!

### Important Notes

 * libcontainer/cgroups users who want to manage cgroup devices need to explicitly
   import libcontainer/cgroups/devices. (#3452, #4248)
 * If building with Go 1.22.x, make sure to use 1.22.4 or a later version.
   (see #4233 for more details)

### Added

 * CI: add actuated-arm64. (#4142)

### Fixed

 * cgroup v2: do not set swap to 0 or unlimited when it's not available. (#4188)
 * Set the default value of CpuBurst to nil instead of 0. (#4210, #4211)

>  * runc exec: fix setting rlimit_nofile. (#4195, #4265, #4282, #4290)

Reviewer: This should not be here since it's also in v1.1.13 but it's too late to fix this. I will remove this line in the github release notes.

>  * libct/cg/fs: fix setting rt_period vs rt_runtime. (#4094, #4258)

Reviewer: Same here. Sorry I overlooked this.

>  * script/*: fix gpg usage wrt keyboxd. (#4189)

Reviewer: same here

>  * Fix a debug log message for user namespaces in nsexec. (#4311)

Reviewer: same here

 * libct/cg: write unified resources line by line. (#4186)
 * libct.Start: fix locking, do not allow a second container init. (#4271)
 * Fix tests in debian testing (mount_sshfs.bats). (#4245)
 * Fix checkpoint/restore tests on actuated-arm64. (#4276)
 * Fix codespell warnings. (#4291)
 * libct/cg/dev: fix TestSetV1Allow panic. (#4295)
 * tests/int/scheduler: require smp. (#4298)

### Changed

>  * libcontainer: allow containers to make apps think fips is enabled/disabled for testing. (#4246)

Reviewer: same here, it's in 1.1.13.

## [Unreleased]

> ## [1.2.0-rc.2] - 2024-06-14

Reviewer: `s/-14/-26/`?

> TRUE or FALSE, it's a problem!

### Important Notes

>  * libcontainer/cgroups users who want to manage cgroup devices need to explicitly

Reviewer: Wonder how this did not made its way into rc1 release notes...

Author: We released rc1 on April 03, but that PR was merged on Apr 26.

Author: Or do you mean remove this `Important Notes` section? If yes, I think this two notes are so important that we should tell to the users. So I add this section.

Reviewer: I asked about #3452, but now I see that #4248 (which comes with this changelog entry) was merged after rc.1. So it's fine.

   import libcontainer/cgroups/devices. (#3452, #4248)
 * If building with Go 1.22.x, make sure to use 1.22.4 or a later version.
   (see #4233 for more details)

### Added

 * CI: add actuated-arm64. (#4142)

### Fixed

 * Fix tests in debian testing (mount_sshfs.bats) - minimal fix. (#4245)
 * Set the default value of CpuBurst to nil instead of 0. (#4211)

>  * Fix set nofile rlimit error. (#4265, #4282, #4290)

Reviewer: I'd say "runc exec: fix setting rlimit_nofile" as the fix is specific for runc exec.

 * script/*: fix gpg usage wrt keyboxd. (#4189)

>  * fix a debug msg for user ns in nsexec. (#4311)

Reviewer 2: not:
```suggestion
 * Fix a debug log message for user namespaces in nsexec. (#4311)
```

>  * libct/cg/fs: fix setting rt_period vs rt_runtime. (#4258)

Reviewer: We usually add issues as well (for this one, it's #4094).

 * Fix a debug log message for user namespaces in nsexec. (#4311)
 * libct/cg: write unified resources line by line. (#4186)
 * libct.Start: fix locking, do not allow a second container init. (#4271)

>  * Fix tests in debian testing (mount_sshfs.bats) - minimal fix. (#4245)

Reviewer 2: maybe like this?
```suggestion
 * Fix tests in debian testing (mount_sshfs.bats). (#4245)
```

>  * fix checkpoint/restore tests on actuated-arm64. (#4276)

Reviewer 2: We can try to start with capital letter always (except when it is runc, a path in the repo, etc.). It seems we've been doing that in the past.

 * Fix checkpoint/restore tests on actuated-arm64. (#4276)
 * Fix codespell warnings. (#4291)
 * libct/cg/dev: fix TestSetV1Allow panic. (#4295)
 * tests/int/scheduler: require smp. (#4298)

>  * tests/int: swap-related fixes. (#4188)

Reviewer 2: Maybe this is better. The other fixes seem to be tests, so this seems the most relevant for the changelog IMHO
```suggestion
 * cgroup v2: do not set swap to 0 or unlimited when it's not available (#4188)
```

 * ci/test: exclude some runc_nodmz jobs. (#4238)
 * libcontainer: force apps to think fips is enabled/disabled for testing. (#4246)

>  * Set temporary single CPU affinity before cgroup cpuset transition.. (#3923)

Reviewer: This one is reverted in https://github.com/opencontainers/runc/pull/4283 and thus should not be listed.

>  * libct: decouple libct/cg/devices. (#4248)

Reviewer 2: I'd remove this line, the changelog entry for this PR has alredy been added here: https://github.com/opencontainers/runc/pull/4317/files#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4edR15

>  * Set temporary single CPU affinity before cgroup cpuset transition. (#3923)

Reviewer 3: this was reverted in #4283 and should **not** be released if the strategy is changing. It would be good to confirm here, @kolyshkin.

Reviewer 2: Grr, yes, this was noted before and asked to remove it completely. Thanks for catching it, github has not the best interface to see if comments where addressed.

@lifubang can you remove it, please? :)

Author: Thanks, I don't know why I didn't remove this. I think I have removed the both lines suggested by @kolyshkin in https://github.com/opencontainers/runc/pull/4317#discussion_r1634205864 
But...

>    Note: this change has been reverted by #4283, it will be implemented by a new way.

Reviewer: Ah I see it now! Maybe just drop both lines.

Reviewer 2: +1 I think it is better to not mention this here.

 * libct/cg/fs: don't write cpu_burst twice on ENOENT. (#4259)
 * allow overriding VERSION value in Makefile. (#4269)
 * make trimpath optional. (#3908)
 * Remove unused system.Execv. (#4268)
 * Stop blacklisting Go 1.22+, drop Go < 1.21 support, use Go 1.22 in CI. (#4292)

>  * Improve some err msgs for runc exec. (#4320)

Reviewer 2: We can expand this too.
``` suggestion
 * Improve some error messages for runc exec. (#4320)
```

[1.1.0-rc.1]: https://github.com/opencontainers/runc/compare/v1.0.0...v1.1.0-rc.1

<!-- 1.2.z patch releases -->

> [1.2.0-rc.2]: https://github.com/opencontainers/runc/compare/v1.2.0-rc.1...v1.2.0-rc.1

Reviewer: ```suggestion
https://github.com/opencontainers/runc/compare/v1.2.0-rc.1...v1.2.0-rc.2
```
